lib/Wikia/tests/Service/User/Auth/MediaWikiAuthServiceTest.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Wikia\Service\User\Auth;
+
+use Wikia\Service\Helios\ClientException;
+use Wikia\Service\Helios\HeliosClient;
+
+class MediaWikiAuthServiceTest extends \WikiaBaseTest {
+	private $authService;
+	private $heliosClientMock;
+
+	public function setUp() {
+		parent::setUp();
+
+		$this->heliosClientMock = $this->getMock( HeliosClient::class,
+			[ 'login', 'forceLogout', 'invalidateToken', 'register', 'info', 'generateToken' ] );
+
+		$this->authService = new MediaWikiAuthService( $this->heliosClientMock );
+	}
+
+	public function testAuthenticateAuthenticationFailed() {
+		$username = 'SomeName';
+		$password = 'Password';
+
+		$this->heliosClientMock->expects( $this->once() )
+			->method( 'login' )
+			->with( $username, $password )
+			->willReturn( [ \WikiaResponse::RESPONSE_CODE_OK, new \StdClass ] );
+
+		$resultData = $this->authService->authenticate( $username, $password );
+		$this->assertFalse( $resultData->success() );
+	}
+
+	public function testAuthenticateAuthenticationImpossible() {
+		$this->setExpectedException( 'Wikia\Service\Helios\ClientException', 'test' );
+		$username = 'SomeName';
+		$password = 'Password';
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Remove this hard-coded password. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS2068)